plugin/pkg/proxy/connect.go

```diff
@@ -117,6 +118,20 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 	for {
 		ret, err = pc.c.ReadMsg()
 		if err != nil {
+			// If the error is an overflow, we probably have a misbehaving upstream
+			// that is sending out-of-spec (>512 byte) UDP responses without an eDNS0
+			// OPT RR. Set the truncate bit and return the result - this will make the
```

Comment:
```suggestion
			// If the error is an overflow, we probably have an upstream misbehaving in some way
			// (e.g. sending >512 byte UDP responses without an eDNS0 OPT RR).
			// Instead of returning an error, return an empty response with TC bit set. This will make the
```